### PR TITLE
[browser] Fix encoding problem when publishing with AOT

### LINKED ISSUE #83035: [browser] Fix encoding problem when publishing with AOT
Follow-up for https://github.com/dotnet/runtime/pull/82833. AOT case.

- Add AOT tests the same way as in the previous PR
- Fix `wasm-ld : error : /root/helix/work/workitem/e/wbt/pztditip.fs4/obj/Debug/net8.0/browser-wasm/wasm/for-publish/driver.o: undefined symbol: mono_aot_module_build_publish_Debug煉_info `

### PR DESCRIPTION
**Pull request by ilonatommy**
Fixes https://github.com/dotnet/runtime/issues/83035.

- In native code we are checking if char is alphanumeric, if not we are replacing it with '_'. Unicode signs are treated as non-alphanumeric. To remain consistent, the same mechanism was added in the managed code, in `FixupSymbolName`.
- We need to block the test part where we check relinking due to https://github.com/dotnet/runtime/issues/83497.

### CODE REVIEW

File: src/mono/wasm/Wasm.Build.Tests/BuildPublishTests.cs [hunk 1]
```diff
@@ -73,7 +73,7 @@ public void BuildThenPublishNoAOT(BuildArgs buildArgs, RunHost host, string id)
         [BuildAndRun(host: RunHost.Chrome, aot: true, config: "Debug")]
         public void BuildThenPublishWithAOT(BuildArgs buildArgs, RunHost host, string id)
         {
-            string projectName = $"build_publish_{buildArgs.Config}";
+            string projectName = $"build_publish_{buildArgs.Config}_{s_unicodeChar}";
```
Comment (kg) [inline src/mono/wasm/Wasm.Build.Tests/BuildPublishTests.cs:76]: Maybe add a comment explaining why s_unicodeChar being here is important

File: src/mono/wasm/Wasm.Build.Tests/BuildPublishTests.cs [hunk 2]
```diff
@@ -108,23 +108,25 @@ public void BuildThenPublishWithAOT(BuildArgs buildArgs, RunHost host, string id
 
             _testOutput.WriteLine($"{Environment.NewLine}Publishing with no changes ..{Environment.NewLine}");
 
+            // relinking for paths with unicode does not work:
+            // [ActiveIssue("https://github.com/dotnet/runtime/issues/83497")]
```
Comment (kg) [inline src/mono/wasm/Wasm.Build.Tests/BuildPublishTests.cs:112]: Might be nice to have a FIXME here so it's easier to find by searching the codebase for them, but I see that the activeissue mentions this method needs to be updated, so it's not obligatory

**Review by kg — APPROVED** [review]

File: src/mono/wasm/Wasm.Build.Tests/BuildPublishTests.cs (hunk 1 above)
Comment (ilonatommy) [reply src/mono/wasm/Wasm.Build.Tests/BuildPublishTests.cs]: Thank you, added (in the common function for creating the test project name).

**Comment by ilonatommy** [comment]
Failures not related, merging.

**Pull request merged**